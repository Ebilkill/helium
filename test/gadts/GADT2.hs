data X a where
    I :: Int -> X Int
    B :: Bool -> X Bool

f :: X a  -> Int -> Int
f (I i) y = i
f (B b) y = y