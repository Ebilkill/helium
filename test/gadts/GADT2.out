Prelude is up to date
Compiling gadts/GADT2.hs
(6,9): Warning: Variable "y" is not used
(7,6): Warning: Variable "b" is not used
fromList [(f,X a -> Int -> Int)]

Length of errors doesn't match

fromList [(f,X a -> Int -> Int)]
"fromList [(f,X a -> Int -> Int)]"
(7,4): Type error in pattern of function binding
 pattern          : B b
   type           : X Bool
   does not match : X Int 

Compilation failed with 1 error
